packages/nitro-protocol/test/contracts/ERC20AssetHolder/deposit.test.ts

```diff
@@ -75,12 +77,13 @@ describe('deposit', () => {
       const {data: amountTransferred} = getTransferEvent(events);
       expect(held.eq(amountTransferred)).toBe(true);
     }
+
+    const balanceBefore = BigNumber.from(await Token.balanceOf(signer0Address));
     const tx = ERC20AssetHolder.deposit(destination, expectedHeld, amount);
```

Comment:
The test is now passing, I think because I put the check on `balanceBefore` _ahead_ of forming the `deposit` transaction. This avoids a race.

Reply:
(if the `deposit` transaction gets mined before the `balanceOf` call, we actually store `balanceAfter` instead of `balanceBefore`, which explains the test failure).